pkg/authentication/basic/htpasswd.go

```diff
@@ -0,0 +1,96 @@
+package basic
+
+import (
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/logger"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// htpasswdMap represents the structure of an htpasswd file.
+// Passwords must be generated with -B for bcrypt or -s for SHA1.
+type htpasswdMap struct {
+	users map[string]interface{}
+}
+
+// bcryptPass is used to identify bcrypt passwords in the
+// htpasswdMap users.
+type bcryptPass string
+
+// sha1Pass os used to identify sha1 passwords in the
+// htpasswdMap users.
+type sha1Pass string
```

Comment:
When do you think is a good time to deprecate SHA1 htpasswd entries (and only allow bcrypt)? We obviously aren't looking for full compatibility with Htpasswd since we don't support MD5 - so this might be the next logical format to drop support for due to security reasons.

Reply:
That's an interesting question, given users have an alternate option and we aren't forcing them to use this feature, I wonder if we need to deprecate it? Perhaps we do so in v7/v8? Personally I think this would be a lower priority deprecation.

How do people migrate from sha1 to bcrypt? 🤔

Reply:
Yeah - agreed: low priority.  I just saw this in a a Gosec report a few weeks back and wanted to make note of it.

No way for a user to migrate unless they knew the original password.

Reply:
Perhaps create an issue for the deprecation and see if we gain any feedback from the community on it?

> No way for a user to migrate unless they knew the original password.

We could always rewrite the htpasswd files as people log in :trollface: